src/model/test/HeatExchangerDesiccantBalancedFlow_GTest.cpp

```diff
@@ -0,0 +1,201 @@
+*  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+*  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***********************************************************************************************************************/
+
+#include <gtest/gtest.h>
+#include "ModelFixture.hpp"
+
+#include "../HeatExchangerDesiccantBalancedFlow.hpp"
+#include "../HeatExchangerDesiccantBalancedFlow_Impl.hpp"
+
+#include "../HeatExchangerDesiccantBalancedFlowPerformanceDataType1.hpp"
+#include "../HeatExchangerDesiccantBalancedFlowPerformanceDataType1_Impl.hpp"
+#include "../Schedule.hpp"
+#include "../Schedule_Impl.hpp"
+#include "../ScheduleConstant.hpp"
+#include "../ScheduleConstant_Impl.hpp"
+#include "../Node.hpp"
+#include "../Node_Impl.hpp"
+#include "../AirLoopHVAC.hpp"
+#include "../AirLoopHVAC_Impl.hpp"
+#include "../AirLoopHVACOutdoorAirSystem.hpp"
+#include "../AirLoopHVACOutdoorAirSystem_Impl.hpp"
+#include "../HVACTemplates.hpp"
+
+using namespace openstudio;
+using namespace openstudio::model;
+
+TEST_F(ModelFixture, HeatExchangerDesiccantBalancedFlow_HeatExchangerDesiccantBalancedFlow) {
+  ::testing::FLAGS_gtest_death_test_style = "threadsafe";
+
+  ASSERT_EXIT(
+    {
+      Model m;
+      HeatExchangerDesiccantBalancedFlow hx(m);
+      HeatExchangerDesiccantBalancedFlowPerformanceDataType1 p(m);
+      HeatExchangerDesiccantBalancedFlow hx2(m, p);
+
+      exit(0);
+    },
+    ::testing::ExitedWithCode(0), "");
+
+  Model m;
+  HeatExchangerDesiccantBalancedFlow hx(m);
+  HeatExchangerDesiccantBalancedFlowPerformanceDataType1 p(m);
+  HeatExchangerDesiccantBalancedFlow hx2(m, p);
+
+  auto alwaysOn = m.alwaysOnDiscreteSchedule();
+
+  EXPECT_EQ(alwaysOn, hx.availabilitySchedule());
+  ASSERT_TRUE(hx.heatExchangerPerformance().optionalCast<HeatExchangerDesiccantBalancedFlowPerformanceDataType1>());
+  EXPECT_FALSE(hx.economizerLockout());
+
+  EXPECT_EQ(alwaysOn, hx2.availabilitySchedule());
+  ASSERT_TRUE(hx2.heatExchangerPerformance().optionalCast<HeatExchangerDesiccantBalancedFlowPerformanceDataType1>());
+  EXPECT_FALSE(hx2.economizerLockout());
+}
+
+TEST_F(ModelFixture, HeatExchangerDesiccantBalancedFlow_GettersSetters) {
+  Model m;
+  HeatExchangerDesiccantBalancedFlowPerformanceDataType1 p(m);
+  HeatExchangerDesiccantBalancedFlow hx(m, p);
+
+  Schedule schedule = hx.availabilitySchedule();
+  boost::optional<ScheduleConstant> scheduleConstant = schedule.optionalCast<ScheduleConstant>();
+  ASSERT_TRUE(scheduleConstant);
+  EXPECT_EQ((*scheduleConstant).value(), 1.0);
+  ASSERT_TRUE(scheduleConstant);
+  HeatExchangerDesiccantBalancedFlowPerformanceDataType1 performanceObject = hx.heatExchangerPerformance();
+  EXPECT_EQ(performanceObject.name().get(), p.name().get());
+
+  ScheduleConstant scheduleConstant2(m);
+  scheduleConstant2.setValue(0.5);
+  hx.setAvailabilitySchedule(scheduleConstant2);
```

Comment:
Same idea. Just assign the new Schedule, then test equality (on handle) instead of relying on value. Shorter, simpler, more accurate.